### DISCUSSION TITLE
Issue with auto-loading helpers

### DISCUSSION THREAD

**Discussion by tswagger** [codeigniter4/shield#974] [General]
I am seeing an issue which I think has more to do with the core CI4 repository than it does Shield, but it is directly effecting Shield.  When adding the "recommended" helpers to \Config\Autoload.php `public $helpers = ['auth', 'setting'];` it seams to have no effect.  I get errors that `auth()` and `setting()` cannot be found.

The only thing that seems to solve the problem is adding the following the the `$files` array in the Autoload.php file:

```php
public $files = [
	'vendor/codeigniter4/shield/src/Helpers/auth_helper.php',
	'vendor/codeigniter4/settings/src/Helpers/setting_helper.php'
];

public $helpers = ['auth', 'setting'];
```

The documentation on this topic imply that I should not have to do this as these helpers are located in a known name space.  This should obviously be addressed in the core CI4 repository (either in code or in the documentation), but my question is should we update the Shield documentation in the mean time?

**Comment by datamweb** [comment]
I don't see a problem at the moment.
So my question is how can I reproduce your problem?
What version of CI are you using, how did you install it, did you follow the steps to upgrade CI correctly?

**Reply by tswagger** [reply]
CI 4.4.3 installed via composer.  All upgrade instructions followed closely.

The project is in early days, so the CI functionality is "out of the box".  I am using the HMAC feature of shield, and the errors are being thrown from the HMAC filter as the request is being authenticated.

Errors were discovered while running unit tests on the controllers using the following type of command:

```php
$result = $this->withHeaders([
	'Authorization' => $authHeader,
	'Content-Type'  => 'application/json',
	'Accept'        => 'application/json',
])
	->withBody($body)
	->post('/myEndpoint');
```

**Reply by datamweb** [reply]
It seems that you are affected by bug number [PR 8275](https://github.com/codeigniter4/CodeIgniter4/pull/8275).
To temporarily fix the test failure, you can load the helpers in the unit test file.

**Reply by tswagger** [reply]
That sounds exactly like my issue.  Thank you. I will close the discussion.
